### PR TITLE
DBAAS1-83: Database Connection Details

### PR DESCRIPTION
**Pull request by HanaXu**
## Description
Adds Connection Details component to the Database Summary tab

## How to test
- Turn on the MSW and go to a database (e.g. `databases/mysql/0`)
- Test the show and download buttons.
    - The download button should only be visible if `ssl_connection` is enabled
    - Test loading and error states

### CODE REVIEW

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseSummary/DatabaseSummaryConnectionDetails.tsx [hunk 1]
```diff
@@ -1,18 +1,205 @@
...
+      .catch((errorResponse: any) => {
+        const error = getErrorStringOrDefault(
+          errorResponse,
+          'Unable to download your CA Certificate'
+        );
+        enqueueSnackbar(error, { variant: 'error' });
+      });
+  };
+
+  if (credentialsLoading) {
```
Comment (bnussman) [inline packages/manager/src/features/Databases/DatabaseDetail/DatabaseSummary/DatabaseSummaryConnectionDetails.tsx:119]: Can we make a big change where the component lazy loads the credentials? I don't think loading the credentials should block the entire component from displaying other relevant data.

**Comment by bnussman** [comment]
Looks like we'll need a dark mode fix
Screen Shot 2022-01-20 at 3 20 30 PM

**Comment by bnussman** [comment]
The box looks very _white_... Maybe we can use `theme.cmrBorderColors.borderTable` for the border and or background? I saw @DevDW use that color for a border

Screen Shot 2022-01-20 at 4 52 17 PM

**Review by DevDW — APPROVED** [review]
LGTM, let's keep an eye on the API side with regards to the `hosts` aspect because the page crashes when the mocks aren't on as the `hosts` properties are undefined.

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseSummary/DatabaseSummaryConnectionDetails.tsx [hunk 2]
```diff
@@ -1,18 +1,230 @@
...
+          ) : (
+            <Typography style={{ alignSelf: 'center' }}>
+              {credentialsError ? (
+                <>
+                  <span className={classes.error}>
+                    Error retrieving credentials.
+                  </span>
+                  {credentialsBtn(() => getDatabaseCredentials(), 'Retry')}
+                </>
+              ) : (
```
Comment (codyfinn) [inline packages/manager/src/features/Databases/DatabaseDetail/DatabaseSummary/DatabaseSummaryConnectionDetails.tsx:179]: Something that can be addressed later but I'm not a big fan of nested ternaries

**Review by codyfinn — APPROVED** [review]
Looks good to me

**Review by bnussman — APPROVED** [review]

**Pull request merged**